File: backend/cmd/server/dtos/analytics.go
package dtos

type SeedAnalytics struct {
	Seed                 int     `json:"seed"`
	TotalPoints          float64 `json:"totalPoints"`
	TotalInvestment      float64 `json:"totalInvestment"`
	PointsPercentage     float64 `json:"pointsPercentage"`
	InvestmentPercentage float64 `json:"investmentPercentage"`
	TeamCount            int     `json:"teamCount"`
	AveragePoints        float64 `json:"averagePoints"`
	AverageInvestment    float64 `json:"averageInvestment"`
	ROI                  float64 `json:"roi"`
}

type RegionAnalytics struct {
	Region               string  `json:"region"`
	TotalPoints          float64 `json:"totalPoints"`
	TotalInvestment      float64 `json:"totalInvestment"`
	PointsPercentage     float64 `json:"pointsPercentage"`
	InvestmentPercentage float64 `json:"investmentPercentage"`
	TeamCount            int     `json:"teamCount"`
	AveragePoints        float64 `json:"averagePoints"`
	AverageInvestment    float64 `json:"averageInvestment"`
	ROI                  float64 `json:"roi"`
}

type TeamAnalytics struct {
	SchoolID          string  `json:"schoolId"`
	SchoolName        string  `json:"schoolName"`
	TotalPoints       float64 `json:"totalPoints"`
	TotalInvestment   float64 `json:"totalInvestment"`
	Appearances       int     `json:"appearances"`
	AveragePoints     float64 `json:"averagePoints"`
	AverageInvestment float64 `json:"averageInvestment"`
	AverageSeed       float64 `json:"averageSeed"`
	ROI               float64 `json:"roi"`
}

type SeedVarianceAnalytics struct {
	Seed             int     `json:"seed"`
	InvestmentStdDev float64 `json:"investmentStdDev"`
	PointsStdDev     float64 `json:"pointsStdDev"`
	InvestmentMean   float64 `json:"investmentMean"`
	PointsMean       float64 `json:"pointsMean"`
	InvestmentCV     float64 `json:"investmentCV"`
	PointsCV         float64 `json:"pointsCV"`
	TeamCount        int     `json:"teamCount"`
	VarianceRatio    float64 `json:"varianceRatio"`
}

type AnalyticsResponse struct {
	SeedAnalytics         []SeedAnalytics         `json:"seedAnalytics,omitempty"`
	RegionAnalytics       []RegionAnalytics       `json:"regionAnalytics,omitempty"`
	TeamAnalytics         []TeamAnalytics         `json:"teamAnalytics,omitempty"`
	SeedVarianceAnalytics []SeedVarianceAnalytics `json:"seedVarianceAnalytics,omitempty"`
	TotalPoints           float64                 `json:"totalPoints"`
	TotalInvestment       float64                 `json:"totalInvestment"`
	BaselineROI           float64                 `json:"baselineROI"`
}

File: frontend/src/types/analytics.ts
export interface SeedAnalytics {
  seed: number;
  totalPoints: number;
  totalInvestment: number;
  pointsPercentage: number;
  investmentPercentage: number;
  teamCount: number;
  averagePoints: number;
  averageInvestment: number;
  roi: number;
}

export interface RegionAnalytics {
  region: string;
  totalPoints: number;
  totalInvestment: number;
  pointsPercentage: number;
  investmentPercentage: number;
  teamCount: number;
  averagePoints: number;
  averageInvestment: number;
  roi: number;
}

export interface TeamAnalytics {
  schoolId: string;
  schoolName: string;
  totalPoints: number;
  totalInvestment: number;
  appearances: number;
  averagePoints: number;
  averageInvestment: number;
  averageSeed: number;
  roi: number;
}

export interface SeedVarianceAnalytics {
  seed: number;
  investmentStdDev: number;
  pointsStdDev: number;
  investmentMean: number;
  pointsMean: number;
  investmentCV: number;
  pointsCV: number;
  teamCount: number;
  varianceRatio: number;
}

export interface SeedInvestmentPoint {
  seed: number;
  tournamentName: string;
  tournamentYear: number;
  calcuttaId: string;
  teamId: string;
  schoolName: string;
  totalBid: number;
  calcuttaTotalBid: number;
  normalizedBid: number;
}

export interface SeedInvestmentSummary {
  seed: number;
  count: number;
  mean: number;
  stdDev: number;
  min: number;
  q1: number;
  median: number;
  q3: number;
  max: number;
}

export interface SeedInvestmentDistributionResponse {
  points: SeedInvestmentPoint[];
  summaries: SeedInvestmentSummary[];
}

export interface BestInvestment {
  tournamentName: string;
  tournamentYear: number;
  calcuttaId: string;
  teamId: string;
  schoolName: string;
  seed: number;
  region: string;
  teamPoints: number;
  totalBid: number;
  calcuttaTotalBid: number;
  calcuttaTotalPoints: number;
  investmentShare: number;
  pointsShare: number;
  rawROI: number;
  normalizedROI: number;
}

export interface BestInvestmentsResponse {
  investments: BestInvestment[];
}

export interface AnalyticsResponse {
  seedAnalytics?: SeedAnalytics[];
  regionAnalytics?: RegionAnalytics[];
  teamAnalytics?: TeamAnalytics[];
  seedVarianceAnalytics?: SeedVarianceAnalytics[];
  totalPoints: number;
  totalInvestment: number;
  baselineROI: number;
}
